mcs/class/corlib/System.Threading/Timer.cs

```diff
@@ -447,6 +441,23 @@ static void TimerCB (object o)
 				return ms_wait;
 			}
 
+			private void MaybeReSort (TimerComparer comparer) {
+				if (needReSort) {
+					try {
+					} finally {
```

Comment:
I think we can just catch the `InvalidOperationException` instead of using the `finally`?

Reply:
Yea this makes sense to me: we _know_ the comparer is a TimerComparer and it doesn't throw anything itself.  So we could just catch and ignore the IOE.